man/udevadm.xml

```diff
@@ -382,6 +382,21 @@
             then each matching result is ORed, that is, all children of each specified device are triggered.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><option>--database-match</option></term>
+          <listitem>
+            <para>Trigger devices with a corresponding udev database entry only. This can potentially save a
+            significant amount of time compared to re-triggering all devices in the system and e.g. can be used to
+            optimize boot time.</para>
```

Comment:
From reading your PR comment, it seems that --database-nomatch is the important one, and that --database-match is here for completeness but is not really usefull.
The way this is worded seems to suggest the opposit.
Also, an expanded description and/or a service infrastructure using this feature would be very welcomed.

Reply:
> From reading your PR comment, it seems that --database-nomatch is the important one, and that --database-match is here for completeness but is not really usefull. The way this is worded seems to suggest the opposit.

Oops, fixed it up. However, this is replaced by https://github.com/systemd/systemd/pull/22662.

> Also, an expanded description and/or a service infrastructure using this feature would be very welcomed.

Probably in the common distributions this wouldn't do much applying it to systemd-udev-trigger.service, since it's executed pretty close after systemd-udevd.

However, there are lots of embedded use-cases, e.g. automotive control units where services can have strong boot time requirements. Blanketly, re-triggering all devices, processing the particular udev rules and potentially forking off and executing subsequent processes is too costly. Therefore it might make sense to just ramp up the minimal needed infrastructure, re-trigger devices needed for such services having tough boot time requirements only and re-trigger everything else in a later stage. See also https://github.com/systemd/systemd/pull/22662#issuecomment-1061076384.

Reply:
Yes, i'm from the embedded world too and indeed the udev infrastructure is usuallly ~50% of the boot time. So any gain is really usefull

my point is that this is not trivial to understand or use, but it's really needed. thus I think a better documentation on how to use this with custom boot optimisation would be usefull. But let's move to the other PR, if that's where the action is...